src/main/java/org/opentripplanner/street/search/state/State.java

```diff
@@ -436,6 +437,17 @@ public boolean containsModeCar() {
     return false;
   }
 
+  /**
+   * Check all edges is traversed on foot {@code mode=WALK}.
+   * <p>
+   * DO NOT USE THIS IN ROUTING IT WILL NOT PERFORM WELL!
+   */
+  public boolean containsModeWalkOnly() {
+    return Stream
```

Comment:
Since this is going to run against many states I think it pays off to optimise by stopping the iteration early if a non-walking mode has been encountered.